File: demo/functions/hello.php
<?php
// hello.php — ChadStart PHP function
// Returns a greeting from the PHP runtime.
//
// Runtime: php
// Trigger: GET /api/fn/greet/php (public)

function handler(array $event, array $ctx): array {
    $name = $event['query']['name'] ?? 'World';
    return [
        'message' => "Hello, {$name}!",
        'runtime' => 'php',
    ];
}

File: demo/functions/hello.rb
# hello.rb — ChadStart Ruby function
# Returns a greeting from the Ruby runtime.
#
# Runtime: ruby
# Trigger: GET /api/fn/greet/ruby (public)

def handler(event, ctx)
  name = ((event['query'] || {})['name'] || 'World')
  { 'message' => "Hello, #{name}!", 'runtime' => 'ruby' }
end
